src/Surfnet/Stepup/IdentifyingData/EventSourcing/IdentifyingDataDecorator.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\IdentifyingData\EventSourcing;
+
+use Broadway\Domain\AggregateRoot;
+use Broadway\EventSourcing\EventSourcingRepository;
+use Broadway\Repository\RepositoryInterface;
+use Surfnet\Stepup\Exception\DomainException;
+use Surfnet\Stepup\IdentifyingData\Api\IdentifyingDataHolder;
+use Surfnet\Stepup\IdentifyingData\Entity\IdentifyingDataRepository;
+
+class IdentifyingDataDecorator implements RepositoryInterface
+{
+    /**
+     * @var EventSourcingRepository
+     */
+    private $aggregateRootRepository;
+
+    /**
+     * @var IdentifyingDataRepository
+     */
+    private $identifyingDataRepository;
+
+    public function __construct(
+        EventSourcingRepository $aggregateRootRepository,
+        IdentifyingDataRepository $identifyingDataRepository
+    ) {
+        $this->aggregateRootRepository = $aggregateRootRepository;
+        $this->identifyingDataRepository = $identifyingDataRepository;
+    }
+
+    public function save(AggregateRoot $aggregate)
+    {
+        if (!$aggregate instanceof IdentifyingDataHolder) {
+            throw new DomainException(
+                'The AggregateRoot must implement the \Surfnet\Stepup\Identity\Api\AccessibleIdentifyingData interface '
```

Comment:
Seems you have renamed the interface after writing this message ^^ Also, isn't this an invalid argument rather than something that doesn't "adhere to a valid data domain"?